### PR TITLE
Snapshot Retention test

### PR DESCRIPTION
**Pull request by susesgartner**
Snapshot retention check

## Problem
We need a test to verify that the snapshot retention limit is respected.
 
## Solution
Created a test that sets the retention limit and snapshot schedule of an existing cluster and then waits for the limit + 1 snapshots to be taken before checking to make sure the retention limit is respected. 
 
## Testing
Local testing and Jenkins testing.

### Manual Testing
Ran the test against an RKE1, RKE2 and K3s cluster

### CODE REVIEW

**Review by slickwarren — APPROVED** [review]
just a nit; overall lgtm

File: go.mod
Comment (markusewalker) [inline go.mod:3]: Why is this added? Just `go 1.22` should be here.

File: go.mod
Comment (markusewalker) [inline go.mod:425]: Since your testing is done with Jenkins, please update this to not point to your forked repo.

File: go.sum
Comment (markusewalker) [inline go.sum:1801]: This will be resolved once you run `go mod tidy` after not pointing to your forked repo.

File: tests/v2/validation/snapshot/snapshot.go
Comment (markusewalker) [inline tests/v2/validation/snapshot/snapshot.go:432]: `rke` can be put into a const block as well.

**Review by markusewalker — COMMENTED** [review]
Few nits, but overall looks good. Will approve once shepherd PR gets merged. Also, once shepherd PR gets merged and this is approved, please open the backport PRs.

File: go.mod
Comment (susesgartner) [reply go.mod:3]: This is due to this [issue](https://github.com/bazelbuild/rules_go/issues/3850)

Comment (susesgartner) [reply go.mod:3]: I will remove the go.mod and go.sum from my review.

File: go.mod
Comment (susesgartner) [reply go.mod:425]: Will do.

File: tests/v2/validation/snapshot/snapshot.go
Comment (susesgartner) [reply tests/v2/validation/snapshot/snapshot.go:431]: I am working on another PR soon for consts where I expand the extensions/defaults in shepherd. I will clean up this const (as well as other files) in that PR.

**Review by slickwarren — COMMENTED** [review]
lgtm; awaiting fresh jenkins run for approval

File: tests/v2/validation/snapshot/snapshot.go
Comment (susesgartner) [reply tests/v2/validation/snapshot/snapshot.go:432]: This will be handled in a the follow PR to https://github.com/rancher/shepherd/pull/107

**Review by markusewalker — APPROVED** [review]
LGTM

**Review by Josh-Diamond — APPROVED** [review]

**Review by slickwarren — APPROVED** [review]

**Pull request merged**
